### PR TITLE
Feat/plus icon shortcut

### PR DESCRIPTION
**Pull request by jchirch**
## Type of Change
- [x] feature ⛲
- [x] documentation update 📃
- [x] styling 🎨
- [x] testing 🧪

## Description
This adds functionality to the plus sign on the menu bar and adds a dropdown menu with links to create new resources (Contacts, Companies, Job Applications).

## Motivation and Context
By adding shortcuts to the plus icon, users will be able to swiftly create new resources for job hunting upon logging in. This reduces the need for users to have to navigate deeper in the app than necessary.
All styling was done to closely match the mockup, but without an official style guide, minor styling refactoring may be necessary in the future. Additionally, it allows developers in the immediate future the ability to build their "Create" pages.

## Related Tickets
All tickets surrounding creating a new resource.

## Screenshots (if appropriate):

Screenshot 2024-12-13 at 2 07 22 PM

## Added Test?
- [x] Yes 🫡
  - E2E Cypress
- [ ] No 🙅
- [x] All previous tests still pass 🥳

## Checklist:

- [x] My code follows the code style of this project.
- [x]  My change requires a change to the documentation.
- [x]  I have updated the documentation accordingly.

### CODE REVIEW

File: src/components/layout/MenuBar.tsx [hunk 1]
```diff
@@ -32,15 +39,34 @@ function MenuBar() {
...
+            </li>
+            <li className="p-2 hover:bg-gray-200 rounded text-center mb-2">
+              <Link to="/companies/new">Create New Company</Link>
+            </li>
+            <li className="p-2 hover:bg-gray-200 rounded text-center">
+              <Link to="/jobapplications/new">Create New Job Application</Link>
+            </li>
+          </ul>
+        )}
+      </div>
```
Comment (MDelarosa1993) [inline src/components/layout/MenuBar.tsx:68]: Just a suggestion not a pressing fix : When the dropdown is opened or closed, adding a small animation or transition to enhance the user experience.

File: cypress/e2e/landingSpec.cy.js [hunk 2]
```diff
@@ -49,15 +49,38 @@ describe('Landing page after logging in spec', () => {
...
+    })
+    it('should render dropdown menu with link to create new contact', () => {
+      cy.visit('http://localhost:3000/')
+      cy.get('button.m-auto > .m-auto').click()
+      cy.contains('Create New Job Application').should('exist');
+      cy.get('.bg-cyan-600 > :nth-child(3) > a').click()
+      cy.url('match', 'http://localhost:3000/jobapplications/new');
+    })  
+  })
 })
```
Comment (MDelarosa1993) [inline cypress/e2e/landingSpec.cy.js:86]: idk if i'm missing this in the tests but you are testing for the presence of dropdown elements, may be worth testing the isDropDownOpen state directly just to check it's being toggled as expected. I might of missed it in your code. Correct me if i'm wrong.

File: src/components/layout/MenuBar.tsx (hunk 1 above)
Comment (jchirch) [reply src/components/layout/MenuBar.tsx:68]: I completely agree, I attempted to find something on tailwind, but was unable to get a transition to work.
This is the tailwind code I attempted, but was unable to get a result.
``className={`bg-cyan-600 m-4 shadow-md rounded-md transition-all duration-300 ease-in-out transform ${
      isDropDownOpen ? 'scale-100 opacity-100' : 'scale-95 opacity-0'``

Comment (MDelarosa1993) [reply src/components/layout/MenuBar.tsx:68]: Does tailwind offer an invisible and visible style where you can always render the elements but conditionally rendering the styles?

File: cypress/e2e/landingSpec.cy.js [hunk 3]
```diff
@@ -49,15 +49,38 @@ describe('Landing page after logging in spec', () => {
...
+      cy.url('match', 'http://localhost:3000/contacts/new');
+    })
+    it('should render dropdown menu with link to create new company', () => {
+      cy.visit('http://localhost:3000/')
+      cy.get('button.m-auto > .m-auto').click()
+      cy.contains('Create New Company').should('exist');
+      cy.get('.bg-cyan-600 > :nth-child(2) > a').click()
+      cy.url('match', 'http://localhost:3000/companies/new');
+    })
+    it('should render dropdown menu with link to create new contact', () => {
```
[the remark this answers is not in the document]
Comment (jchirch) [reply cypress/e2e/landingSpec.cy.js:78]: Good catch, thanks. Fixed.

**Comment by jimmacur** [comment]
That new transition is looking smooth!  

The latest tests are all failing - The conversion in the menu bar from the old google links to the actual site navigation testing needs updating and it looks like some of the drop tests need updating to the new MuiSvgIcons.

**Review by jimmacur — APPROVED** [review]

**Pull request merged**